File: Examples/CSharp/PivotTable/RefreshAndCalculateItems.cs
using System.IO;
using Aspose.Cells.Pivot;
using Aspose.Cells;
using System.Drawing;

namespace Aspose.Cells.Examples.CSharp.PivotTableExamples
{
    public class RefreshAndCalculateItems
    {
        public static void Run()
        {
            // ExStart:1
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);

            // Load source excel file containing a pivot table having calculated items
            Workbook wb = new Workbook(dataDir + "sample.xlsx");

            // Access first worksheet
            Worksheet sheet = wb.Worksheets[0];

            // Change the value of cell D2
            sheet.Cells["D2"].PutValue(20);

            // Refresh and calculate all the pivot tables inside this sheet
            foreach (PivotTable pt in sheet.PivotTables)
            {
                pt.RefreshData();
                pt.CalculateData();
            }

            // Save the workbook in output pdf
            wb.Save(dataDir + "RefreshAndCalculateItems_out.pdf", SaveFormat.Pdf);
            // ExEnd:1

        }
    }
}

File: Examples/VisualBasic/PivotTable/RefreshAndCalculateItems.vb
Imports System.IO
Imports Aspose.Cells.Pivot
Imports Aspose.Cells
Imports System.Drawing

Namespace PivotTableExamples
    Public Class RefreshAndCalculateItems
        Public Shared Sub Run()
            ' ExStart:1
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            ' Load source excel file containing a pivot table having calculated items
            Dim wb As New Workbook(dataDir & Convert.ToString("sample.xlsx"))

            ' Access first worksheet
            Dim sheet As Worksheet = wb.Worksheets(0)

            ' Change the value of cell D2
            sheet.Cells("D2").PutValue(20)

            ' Refresh and calculate all the pivot tables inside this sheet
            For Each pt As PivotTable In sheet.PivotTables
                pt.RefreshData()
                pt.CalculateData()
            Next

            ' Save the workbook in output pdf
            wb.Save(dataDir & Convert.ToString("RefreshAndCalculateItems_out.pdf"), SaveFormat.Pdf)
            ' ExEnd:1

        End Sub
    End Class
End Namespace
